polaris-react/src/utilities/event-listener/hooks.ts

```diff
@@ -0,0 +1,21 @@
+import {useEffect, useRef} from 'react';
+
+export type ElementType = Window | Document | HTMLElement;
+
+export function useEventListener(
+  event: string,
+  handler: () => void,
+  element: ElementType = window,
+  options?: AddEventListenerOptions,
+) {
+  const handlerRef = useRef(handler);
+
+  useEffect(() => {
+    const handler = handlerRef.current;
+    element.addEventListener(event, handlerRef.current, options);
+
+    return () => {
+      element.removeEventListener(event, handler, options?.capture);
+    };
+  }, [event, element, options]);
```

Comment:
```suggestion
  }, [event, element]);
```

I believe we should also set this in a `ref` so an unmemorized object does re-run this effect each render.